scan

```diff
@@ -362,7 +361,7 @@ def perform_scan(params):
         # If --meta wasn't requested, throw it all away.
         if not options.get("meta", False): meta = None
 
-        write_rows(rows, handles[scanner]['headers'], domain, scanner, handles[scanner]['writer'], meta=meta)
+        runner.write_rows(rows, domain, utils.base_domain_for(domain), scanner, handles[scanner]['writer'], meta=meta)
```

Comment:
Couple of differences here:
- `write_rows` wasn't using the `headers` argument at all, so I removed it
- `write_rows` was pulling options directly from `argparse`. That's really difficult to test, and makes the function less atomic, so I pulled it out. I imagine we'll see a lot of that as we go.